lib/ProductOpener/Food.pm

```diff
@@ -4411,12 +4424,90 @@ sub compute_serving_size_data($) {
 			}	
 		
 		}
+		
+		# Carbon footprint
+		if (defined $product_ref->{nutriments}{"carbon-footprint-from-meat-or-fish_100g"}) {
+		
+			if (defined $product_ref->{serving_quantity}) {
+				$product_ref->{nutriments}{"carbon-footprint-from-meat-or-fish_serving"}
+				= sprintf("%.2e",$product_ref->{nutriments}{"carbon-footprint-from-meat-or-fish_100g"} / 100.0 * $product_ref->{serving_quantity}) + 0.0;
+			}
+			
+			if (defined $product_ref->{product_quantity}) {
+				$product_ref->{nutriments}{"carbon-footprint-from-meat-or-fish_product"}
+				= sprintf("%.2e",$product_ref->{nutriments}{"carbon-footprint-from-meat-or-fish_100g"} / 100.0 * $product_ref->{product_quantity}) + 0.0;
+			}
+		}
 	
 	}
 
 }
 
 
+sub compute_carbon_footprint_infocard($) {
+
+	my $product_ref = shift;
+	
+	if ((defined $product_ref->{nutriments}) and (defined $product_ref->{nutriments}{"carbon-footprint-from-meat-or-fish_product"})) {
+
+		foreach my $lang ("en", "fr") {
```

Comment:
at this point, it's largely a demo, and it's relying on figures that apply to French products. We'd need to get equivalent figures for Germany.

Reply:
It's difficult to remove a demo from a public API after it has been published. Wouldn't it make more sense to limit it to selling country or producer instead? Also, I don't see any restriction in `compute_carbon_footprint_from_ingredients($)` that would limit the algorithm to French products. Unless I'm mistaken, the carbon footprint should be calculated for any product that has a matching ingredient in the taxonomy with percentages.

Reply:
A good point. Maybe we should compute the carbon values only for products sold in France, and compute the infocard on the fly when the OFF app (and not another app) is asking for the JSON. The infocard thing is mostly for the OFF app. It's easier to generate it server side though (so that we can update it whenever we want without changing the app).